src/analyze/analyze-condition.c

```diff
@@ -92,9 +110,27 @@ int verify_conditions(char **lines, UnitFileScope scope) {
                 return log_error_errno(r, "Failed to create test.service: %m");
 
         STRV_FOREACH(line, lines) {
-                r = parse_condition(u, *line);
-                if (r < 0)
-                        return r;
+                if (unit_name_is_valid(*line, UNIT_NAME_ANY)) {
+                        _cleanup_free_ char *prepared = NULL;
+                        Unit *unit;
+
+                        r = verify_prepare_filename(*line, &prepared);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to prepare filename %s: %m", *line);
+
+                        r = manager_load_startable_unit_or_warn(m, NULL, prepared, &unit);
+                        if (r < 0)
+                                return r;
+
+                        if (unit->asserts)
+                                LIST_JOIN(conditions, u->asserts, unit->asserts);
+                        if (unit->conditions)
+                                LIST_JOIN(conditions, u->conditions, unit->conditions);
```

Comment:
uh, this means the list of conditions will be freed twice.

Reply:
How so? `LIST_JOIN` sets the second passed list to `NULL` so after `LIST_JOIN` the conditions should only be referenced by `u` and not by `unit` anymore (unless I'm missing something).